pyscf/mcpdft/tfnal_derivs.py

```diff
@@ -455,6 +481,34 @@ def _tGGA_jT_op(x, rho, Pi, R, zeta):
     return jTx
 
 
+def _tmetaGGA_jT_op(x, rho, Pi, R, zeta):
+    # output ordering is
+    # ordering: rho, Pi, |rho'|^2, tau
+    ngrid = rho.shape[-1]
+    jTx = np.zeros((4, ngrid), dtype=x[0].dtype)
+    if R.ndim > 1:
+        R = R[0]
+
+    # ab -> cs coordinate transformation
+    xc = (x[5] + x[6]) / 2.0
+    xm = (x[5] - x[6]) / 2.0
+
+    # easy part
+    jTx[3] = xc + zeta[0] * xm
```

Comment:
Is this vulnerable to the problem that #113 solved? We haven't seen it happen yet but is it possible that some functional might return nonsense on x[6] when there are no beta electrons?

Reply:
wait nevermind, thats what 121-129 are for

Reply:
Yes, if I don't have the 121-129 lines to prevent 0 beta electrons, I will get non-sensical $\frac{\partial \epsilon^\mathrm{xc}}{\tau_\downarrow}$. Specifically, this is an issue with the $\frac{\partial}{\partial \Pi}$ term.

Reply:
One could make the argument that we ought to have "defense in depth" for this sort of thing by both making sure we never have identically zero translated beta electrons and by indexing explicitly for zeta being identically one in this function, in case a situation ever arises where 1e-15 is not far enough away from zero to control instability. But IDK if it's that important.

Reply:
Its actually not this step that causes an issue. I have tried indexing off of `zeta==1` and setting the default to `x[5]` but this doesn't solve the problem. The problem, for H2 triplet, tM06L functional, is actually on the `jTx[1]` term, or the derivative with respect to $\Pi$. If I write it out exactly, we have that 

$$
  \frac{2\tau}{\rho^2} \frac{\partial\zeta}{\partial R} \left( \frac{\partial \epsilon^\mathrm{xc}}{\tau_\uparrow} - \frac{\partial \epsilon^\mathrm{xc}}{\tau_\downarrow}\right)
$$

At $R = 0$, $\zeta = 1$ and $\frac{\partial\zeta}{\partial R} = -\frac{1}{2}$. So if $\frac{\partial \epsilon^\mathrm{xc}}{\tau_\downarrow}$ gives non-sensible values because of whatever issue is present in libxc, there is no multiplier out front to 'kill' this term off.